apps/devhub/tasks.py

```diff
@@ -32,65 +32,116 @@
 log = logging.getLogger('z.devhub.task')
 
 
+# Simple wrapper for the sake of tests.
+def validate(file_, listed=None):
+    """Run the validator on the given File or FileUpload object, and annotate
+    the results using the ValidationAnnotator."""
+
+    # Import loop.
+    from .utils import ValidationAnnotator
+
+    return ValidationAnnotator(file_, listed=listed).task.delay()
+
+
 @task(soft_time_limit=settings.VALIDATOR_TIMEOUT)
 @write
-def validator(upload_id, listed=True, **kw):
-    if not settings.VALIDATE_ADDONS:
-        return None
+def validate_upload(upload_id, listed, **kw):
+    """Run the validator against a file at the given path, and return the
+    results."""
+    if settings.VALIDATE_ADDONS:
+        upload = FileUpload.objects.get(pk=upload_id)
+        try:
+            result = run_validator(upload.path, listed=listed)
+        except Exception:
```

Comment:
Seems you lost the comments from the "previous version" https://github.com/mozilla/olympia/pull/610/files#diff-2caa929fd8ca1a356cc34d6108586093L45